### PR TITLE
[CELEBORN-1242] Unify celeborn thread name format

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?

Unify celeborn thread name format with the following pattern:

- client: `celeborn-client-[component]-[function]er`
- service: `[master|worker]-[component]-[function]er`
- other: `celeborn-[component]-[function]er`

### Why are the changes needed?

It's recommended to unify celeborn thread name format especially client side for application.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

No.

### CODE REVIEW

**Review by cxzl25 — APPROVED** [review]

**Comment by SteNicholas** [comment]
@AngersZhuuuu, what's the pattern of the unified thread name?

**Comment by AngersZhuuuu** [comment]
> @AngersZhuuuu, what's the pattern of the unified thread name?

`component-whattodo-xxxer`

File: client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleMaster.java [hunk 1]
```diff
@@ -56,9 +56,8 @@ public class RemoteShuffleMaster implements ShuffleMaster<RemoteShuffleDescripto
   private volatile LifecycleManager lifecycleManager;
   private final ShuffleTaskInfo shuffleTaskInfo = new ShuffleTaskInfo();
   private ShuffleResourceTracker shuffleResourceTracker;
-  private final ScheduledThreadPoolExecutor executor =
-      new ScheduledThreadPoolExecutor(
-          1, ThreadUtils.namedThreadFactory("remote-shuffle-master-executor"));
+  private final ScheduledExecutorService executor =
+      ThreadUtils.newDaemonSingleThreadScheduledExecutor("celeborn-remote-shuffle-master-executor");
```
Comment (SteNicholas) [inline client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleMaster.java:60]:
```suggestion
      ThreadUtils.newDaemonSingleThreadScheduledExecutor("celeborn-client-remote-shuffle-master-executor");
```

**Comment by SteNicholas** [comment]
@AngersZhuuuu, could the pattern of the client thread name be `celeborn-client-component-xxx`, and the pattern of the worker thread name be `worker-component-xxx`?

**Review by SteNicholas — APPROVED** [review]

File: common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java [hunk 2]
```diff
@@ -95,15 +95,15 @@ public TransportResponseHandler(TransportConf conf, Channel channel) {
         if (pushTimeoutChecker == null) {
           pushTimeoutChecker =
               ThreadUtils.newDaemonThreadPoolScheduledExecutor(
-                  "push-timeout-checker", conf.pushDataTimeoutCheckerThreads());
+                  "celeborn-client-push-timeout-checker", conf.pushDataTimeoutCheckerThreads());
```
Comment (FMX) [inline common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java:98]: This thread is used by clients and workers.

Comment (AngersZhuuuu) [reply common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java]: > This thread is used by clients and workers.

How about current?

**Review by FMX — APPROVED** [review]
LGTM. Merged into main(v0.5.0) and branch-0.4(v0.4.1).

**Pull request closed without merging**